src/openms/source/FORMAT/HANDLERS/MzIdentMLHandler.cpp

```diff
@@ -614,6 +614,16 @@ namespace OpenMS
           sid = String(it->getMetaValue("spectrum_id"));
           if (sid.empty())
           {
+            if (emz.empty())
```

Comment:
I think this doesn't work as emz and ert will already  contain the String conversion of nan